node/actors/bft/src/replica/tests.rs

```diff
@@ -1,54 +1,477 @@
+    assert_matches!(
+        res,
+        Err(LeaderPrepareError::InvalidLeader { correct_leader, received_leader }) => {
+            assert_eq!(correct_leader, util.key_at(1).public());
+            assert_eq!(received_leader, util.key_at(0).public());
+        }
+    );
+}
+
+#[tokio::test]
+async fn leader_prepare_old_view() {
+    let mut util = UTHarness::new().await;
+
+    let mut leader_prepare = util
+        .new_procedural_leader_prepare()
+        .await
+        .cast::<LeaderPrepare>()
+        .unwrap()
+        .msg;
+    leader_prepare.view = util.current_replica_view().prev();
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+    assert_matches!(
+        res,
+        Err(LeaderPrepareError::Old { current_view, current_phase }) => {
+            assert_eq!(current_view, util.current_replica_view());
+            assert_eq!(current_phase, util.current_replica_phase());
+        }
+    );
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_sig() {
+    let mut util = UTHarness::new().await;
+
+    let mut leader_prepare = util.new_rnd_leader_prepare(|_| {});
+    leader_prepare.sig = util.rng().gen();
+
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+    assert_matches!(res, Err(LeaderPrepareError::InvalidSignature(..)));
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_prepare_qc() {
+    let mut util = UTHarness::new().await;
+
+    let mut leader_prepare = util
+        .new_procedural_leader_prepare()
+        .await
+        .cast::<LeaderPrepare>()
+        .unwrap()
+        .msg;
+    leader_prepare.justification = util.rng().gen::<PrepareQC>();
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+    assert_matches!(
+    res,
+    Err(LeaderPrepareError::InvalidPrepareQC(err)) => {
+        assert_eq!(err.to_string(), "PrepareQC contains messages for different views!")
+        }
+    )
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_high_qc() {
+    let mut util = UTHarness::new().await;
+
+    let mut replica_prepare = util
+        .new_current_replica_prepare(|_| {})
+        .cast::<ReplicaPrepare>()
+        .unwrap()
+        .msg;
+    replica_prepare.high_qc = util.rng().gen();
+
+    let mut leader_prepare = util
+        .new_procedural_leader_prepare()
+        .await
+        .cast::<LeaderPrepare>()
+        .unwrap()
+        .msg;
+
+    let high_qc = util.rng().gen();
+    leader_prepare.justification = util.new_prepare_qc(|msg| msg.high_qc = high_qc);
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+    assert_matches!(res, Err(LeaderPrepareError::InvalidHighQC(_)))
```

Comment:
Nit: Missing `;` at the end of the statement (ditto in some other tests).

Reply:
Fixed.